tests/test_apiroute.py

```diff
@@ -0,0 +1,39 @@
+from typing import List
```

Comment:
I must confess a little confusion with the test layout. Files seem to be named after possible test scenarios, but there are often multiple files testing the same scenario (e.g. `tests/test_additional_response*`). This also means the unit tests for a given `class` are often spread over many files. Given that, I wasn't sure where to add the new tests, so settled for creating new files. I'm happy to rearrange them, please just LMK how.

Reply:
Yeah, the main tests are for the examples in the docs. And then there are extra tests to cover what is not already covered in those. So, it's indeed quite mixed. But it's fine, I just updated them a bit :slightly_smiling_face: